OrbitLinuxTracing/PerfEvent.h

```diff
@@ -269,13 +264,31 @@ class AbstractUprobesPerfEvent {
   const Function* function_ = nullptr;
 };
 
-class UprobesWithStackPerfEvent : public SamplePerfEvent,
-                                  public AbstractUprobesPerfEvent {
+class UprobesPerfEvent : public PerfEvent, public AbstractUprobesPerfEvent {
  public:
-  explicit UprobesWithStackPerfEvent(uint64_t dyn_size)
-      : SamplePerfEvent{dyn_size} {}
+  perf_event_sp_ip_8bytes_sample ring_buffer_record;
+
+  uint64_t GetTimestamp() const override {
+    return ring_buffer_record.sample_id.time;
+  }
 
   void Accept(PerfEventVisitor* visitor) override;
+
+  pid_t GetPid() const { return ring_buffer_record.sample_id.pid; }
+  pid_t GetTid() const { return ring_buffer_record.sample_id.tid; }
+
+  uint64_t GetStreamId() const {
+    return ring_buffer_record.sample_id.stream_id;
+  }
+
+  uint32_t GetCpu() const { return ring_buffer_record.sample_id.cpu; }
+
+  uint64_t GetSp() const { return ring_buffer_record.regs.sp; }
```

Comment:
Nit: Consider full names like GetStackPointer, GetInstructionPointer or add a comment to indicate what "sp" and "ip" means. This is obvious when you are familiar with the code and context, but for someone reading this for the first time, might not be that obvious.

Reply:
I only added comments, as, while the full names can be better for a first-time reader (I see the point), this is very specific code and the full names can be a bit excessive. After all sp and ip are the actual names of the registers (a bit like "pid").